src/content/en/fundamentals/engage-and-retain/push-notifications/index.markdown

```diff
@@ -1,24 +1,77 @@
+done.
+
+Push notifications are one of the most valuable capabilities of native apps, and
+this capability is now available on the web. To get the most out of them,
+notifications need to be timely, precise, and relevant.
 </p>
 
-In 2015, [a case study](http://googlewebmastercentral.blogspot.com/2015/07/google-case-study-on-app-download-interstitials.html) 
-found that when prompted to download and install Google+, 9% of users did so. 
-That's an impressive click-through rate, or it would be if we were talking
-about online ads. The same study found that 69% of users both left the page 
-that prompted and ignored the invite to install Google+.
+<div class="mdl-grid">
+  <div class="mdl-cell mdl-cell--6-col" markdown="1">
+_Timely_—A timely notification is one that appears when users want it and when
+it matters to them.
+
+_Precise_—A precise notification is one that has specific information that can
+be acted on immediately.
+
+_Relevant_—A relevant message is one about people or subjects the user cares
+about.
+  </div>
+  
+  <figure class="mdl-cell mdl-cell--6-col">
+    <img src="images/tpnr.png" alt="Timely, precise, and relevant">
+    <figcaption>Timely, precise, and relevant</figcaption>
+  </figure>
+</div>
+
+## Are service workers involved? {#service-worker-involved}
+
+Yes. We're not going to explain service workers. We will use service worker code
+in later sections when we show you how to implement pushes and notifications.
+
+## Two technologies {#two-technologies}
+
+Push and notification are different, but complementary, features: a **push** is
+the action of the server supplying information to a service worker; a
+**notification** is the action of a service worker or web page showing the
+information to a user.
+
+## A little notification anatomy {#anatomy}
+
+In the next section we're going to throw a bunch of pictures at you, but we
+promised code. So, here it is. With a service worker registration you call
+`showNotification` on a registration object.
+
+{% highlight javascript %}
+serviceWorkerRegistration.showNotification(title, options);
+{% endhighlight %}
 
+The `title` argument appears as a heading in the notification. The `options`
+argument is an object literal that sets the other properties of a notification.
+A typical options object looks something like this:
 
-Push notifications and the technologies behind them improve retention odds by taking advantage of the code that's already on the user's device: your web page. If you're deploying a [web app manifest](/web/updates/2014/11/Support-for-installable-web-apps-with-webapp-manifest-in-chrome-38-for-Android), or following the principles of [application shell architecture](/web/updates/2015/11/app-shell) then your web pages are already more like native apps than are other web sites. Even if they're not using those technologies, you can retain more users by using push notifications more effectively.
+{% highlight json %}
+{
+  body: 'Are you free tonight?',
```

Comment:
Double quote everything in this JSON for better formatting. Compare how [this looks](https://pr-3031-dot-weasel-dot-web-central.appspot.com/web/fundamentals/engage-and-retain/push-notifications/#anatomy) to when it is [double quoted](https://pr-3031-dot-weasel-dot-web-central.appspot.com/web/fundamentals/engage-and-retain/push-notifications/sending-messages#subscription-anatomy).